### DISCUSSION TITLE
Welcome to sleepme_thermostat Discussions!

### DISCUSSION THREAD

**Discussion by rsampayo** [rsampayo/sleepme_thermostat#10] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ChrisFontanini** [comment]
For the record, the low water sensor does work. The sensor state goes from "Ok" to "Problem".

**Comment by rsampayo** [comment]
Thanks @ChrisFontanini , @mikesalz and I have updated the code. It is now more efficient with the API calls and manages the errors a lot better.

**Reply by mikesalz** [reply]
hmmm... I gave it a try but it is failing.

I get this after a restart:
> 
> Logger: homeassistant.config_entries
> Source: config_entries.py:935
> First occurred: 10:01:27 PM (2 occurrences)
> Last logged: 10:01:27 PM
> 
> Migration handler not found for entry Jamie’s Dock Pro for sleepme_thermostat
> Migration handler not found for entry Mike’s Dock Pro for sleepme_thermostat


And this when I try to add a new entry:
![image](https://github.com/user-attachments/assets/7b6baa33-8567-4da5-b45d-09098d1baafd)

**Reply by ChrisFontanini** [reply]
I updated to the latest version and had the same migration error. I was able to re-add each Dock without issue using a new ApI key

**Reply by mikesalz** [reply]
Using a NEW API key? As in you created a new on rather than using the same one you used previously?

**Reply by ChrisFontanini** [reply]
Yes, I didn't even try with the existing one because I didn't write it down 😂

**Comment by rsampayo** [comment]
Ill look in to that.

**Comment by rsampayo** [comment]
it should work now. you will see This error originated from a custom integration.

Logger: custom_components.sleepme_thermostat.sleepme_api
Source: custom_components/sleepme_thermostat/sleepme_api.py:72
integration: SleepMe Dock Pro (documentation)
First occurred: 8:36:19 PM (4 occurrences)
Last logged: 8:37:56 PM

Previous PATCH request and its retries were cancelled successfully (CancelledError).

 but that is just an warning im using as a debug.  Just check to see if everything is working (at least as good as it was before)

**Reply by mikesalz** [reply]
I'm still getting the same errors this morning.  I even removed both of my Dock Pros and restarted HA. :(

**Reply by rsampayo** [reply]
Can you send me your log?

**Reply by mikesalz** [reply]
Well this is interesting.  Based on @ChrisFontanini's results, I decided to create a new API key.  And it worked.  I'm not sure why, as that doesn't really make sense to me.

**Reply by rsampayo** [reply]
> Well this is interesting.  Based on @ChrisFontanini's results, I decided to create a new API key.  And it worked.  I'm not sure why, as that doesn't really make sense to me. 

@mikesalz , Can you replicate the error with your old api token?

**Reply by mikesalz** [reply]
Unfortunately no.  As soon as I saw the new key was working, I deleted the old one.  But for what it's worth, I didn't see anything useful in the logs prior.

**Comment by rsampayo** [comment]
I guess if the 3 people that uses the code cant replicate the error then we dont have to worry about it. :)

**Comment by mikesalz** [comment]
@rsampayo I didn't want to open a new issue yet because I think is an issue with the API rather than the integration.  I have these in my log from overnight.  But I wanted to share:

```
2024-08-29 00:09:42.510 ERROR (MainThread) [custom_components.sleepme_thermostat.sleepme_api] [GET-devices/zx-cdke5ghn4o7c70mv1khg-1724904581] Exception occurred: Server error '500 Internal Server Error' for url 'https://api.developer.sleep.me/v1/devices/zx-cdke5ghn4o7c70mv1khg'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500. Passing to handle_error.
2024-08-29 00:09:42.510 ERROR (MainThread) [custom_components.sleepme_thermostat.sleepme_api] [GET-devices/zx-cdke5ghn4o7c70mv1khg-1724904582] API request to devices/zx-cdke5ghn4o7c70mv1khg failed after all retries.
2024-08-29 01:13:17.772 ERROR (MainThread) [custom_components.sleepme_thermostat.sleepme_api] [GET-devices/zx-cdp37u96epuc70v4m8u0-1724908391] Exception occurred: . Passing to handle_error.
2024-08-29 01:13:17.773 ERROR (MainThread) [custom_components.sleepme_thermostat.sleepme_api] [GET-devices/zx-cdp37u96epuc70v4m8u0-1724908397] API request to devices/zx-cdp37u96epuc70v4m8u0 failed after all retries.
2024-08-29 01:22:05.687 ERROR (MainThread) [custom_components.sleepme_thermostat.sleepme_api] [GET-devices/zx-cdp37u96epuc70v4m8u0-1724908919] Exception occurred: . Passing to handle_error.
2024-08-29 01:22:05.689 ERROR (MainThread) [custom_components.sleepme_thermostat.sleepme_api] [GET-devices/zx-cdp37u96epuc70v4m8u0-1724908925] API request to devices/zx-cdp37u96epuc70v4m8u0 failed after all retries.
```

**Reply by rsampayo** [reply]
I just checked my log and I also have them at the same time. They are server-side (the 500) but I can probably manage the following ones more gracefully. Open an issue and I'll see what I can do.

**Comment by PloyThought** [comment]
Just have to say people like y'all are amazing! That fact that unpaid projects like this can be shared with people like me who barely know any code, is just incredible 🤩. Thank you all.
